src/screens/Main/EditStack/SnippetSelection/SnippetSelection.tsx

```diff
@@ -20,120 +20,115 @@ interface Props {
-                              orderInList: selectedQuestions[i].orderInList,
-                              text: selectedQuestions[i].text
-                            }
-                            temp.push(newQuestion)
-                          }
-                        } else{
-                          const newQuestion: QuestionSelected = {
-                            id: item.id, 
-                            isSelected: false, 
-                            orderInList: 0,
-                            text: item.text
-                          }
-                          temp.push(newQuestion)
-                        }
-                        setSelectedQuestions(temp)
-                      }
-                    } else{
-                      const temp = []
-                      for(let i = 0; i < selectedQuestions.length; i++){
-                        if(i != (item.id-1)){
-                          const newQuestion: QuestionSelected = {
-                            id: selectedQuestions[i].id, 
-                            isSelected: selectedQuestions[i].isSelected, 
-                            orderInList: selectedQuestions[i].orderInList,
-                            text: selectedQuestions[i].text
-                          }
-                          temp.push(newQuestion)
-                        } else{
-                          const newQuestion: QuestionSelected = {
-                            id: item.id, 
-                            isSelected: true, 
-                            orderInList: numSelected,
-                            text: item.text
-                          }
-                          temp.push(newQuestion)
-                        }
-                        setSelectedQuestions(temp)
-                      }
-                    }
-
-                    if(item.isSelected){
-                      setNumSelected(numSelected-1)
-                      if(numSelected == 2){
-                        setText('A great video requires at least 3 - 4 clips')
-                      } else if(numSelected == 3){
-                        setText((numSelected-2) + ' snippet selected')
-                      } else{
-                        setText((numSelected-2) + ' snippets selected')
-                      }
-                    } else{
-                      setNumSelected(numSelected+1)
-                      if(numSelected == 0){
-                        setText('A great video requires at least 3 - 4 clips')
-                      } else if(numSelected == 1){
-                        setText(numSelected + ' snippet selected')
-                      } else{
-                        setText(numSelected + ' snippets selected')
-                      }
-                    }
+                    updateSnippetState(item, snippetState, selectedSnippetCount, setSnippetState, setSelectedSnippetCount)
+                    updateBottomText(item.isSelected, selectedSnippetCount, setText)
```

Comment:
once you put these functions inside the component, you won't need to pass in anything except item.